### PR TITLE
Support for Dashboard Sharing

### PR DESCRIPTION
**Pull request by shaswot77**
# Description

 Added a new `squaredup_dashboard_share` resource which will allow to enable OA for Dashboards

We already support enabling Open Access for workspace resource. Next will be to allow enabling OA at tenant level.

# Checklist:
- [x] Added a label: `bug-fix`, `feature`, or `enhancement`
- [x] PR description written

### CODE REVIEW

File: internal/provider/resource_open_acess.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+func (r *OpenAccessResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dashboard_share"
+}
+
+func (r *OpenAccessResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Enable Open Access for a dashboard",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the Open Access",
```
Comment (wigsaparelli) [inline internal/provider/resource_open_acess.go:51]: The ID of the Open Access dashboard

File: docs/resources/dashboard_share.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+- `require_authentication` (Boolean) If false, the dashboard will be accessible to anyone with the link
+- `workspace_id` (String) The ID of the workspace where the dashboard is located
+
+### Optional
+
+- `enable_link` (Boolean) If false, the dashboard will not be accessible via Open Access Link
+
+### Read-Only
+
+- `id` (String) The ID of the Open Access
```
Comment (wigsaparelli) [inline docs/resources/dashboard_share.md:79]: The ID of the Open Access dashboard

**Comment by wigsaparelli** [comment]
A general comment:
I think we should avoid using "open access" / "openaccess" etc. anywhere, we don't need to (fine for api endpoints as required) and use shares / sharing / shared dashboard(s). Our user interface never uses the term open access, you only see it in the address bar if you happen to inspect it. The UI always refers to sharing / shares / shared dashboard(s) and I think we should stick to the same nomenclature in Terraform.

File: docs/resources/dashboard_share.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+### Required
+
+- `dashboard_id` (String) The ID of the dashboard
+- `require_authentication` (Boolean) If false, the dashboard will be accessible to anyone with the link
+- `workspace_id` (String) The ID of the workspace where the dashboard is located
+
+### Optional
+
+- `enable_link` (Boolean) If false, the dashboard will not be accessible via Open Access Link
```
Comment (wigsaparelli) [inline docs/resources/dashboard_share.md:75]: I'd simplify to "enabled" and use "(Boolean) If false, sharing of the dashboard is disabled"

File: docs/resources/dashboard_share.md (hunk 2 above)
Comment (shaswot77) [reply docs/resources/dashboard_share.md]: Based on the comment below I have changed it to `The ID of the Shared dashboard`

File: internal/provider/resource_open_acess.go (hunk 1 above)
Comment (shaswot77) [reply internal/provider/resource_open_acess.go]: Based on the comment below I have changed it to The ID of the Shared dashboard

**Comment by shaswot77** [comment]
> A general comment: I think we should avoid using "open access" / "openaccess" etc. anywhere, we don't need to (fine for api endpoints as required) and use shares / sharing / shared dashboard(s). Our user interface never uses the term open access, you only see it in the address bar if you happen to inspect it. The UI always refers to sharing / shares / shared dashboard(s) and I think we should stick to the same nomenclature in Terraform.

Good Point! I have renamed all open access to shared dashboard.

File: docs/resources/dashboard_share.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "squaredup_dashboard_share Resource - terraform-provider-squaredup"
+subcategory: ""
+description: |-
+  Enable Sharing for a dashboard
+---
+
+# squaredup_dashboard_share (Resource)
+
+Enable Sharing for a dashboard
```
Comment (wigsaparelli) [inline docs/resources/dashboard_share.md:11]: sharing

File: docs/resources/dashboard_share.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+- `workspace_id` (String) The ID of the workspace where the dashboard is located
+
+### Optional
+
+- `enabled` (Boolean) If false, sharing of the dashboard is disabled
+
+### Read-Only
+
+- `dashboard_share_link` (String) Shareable link for the dashboard
+- `id` (String) The ID of the Shared dashboard
```
Comment (wigsaparelli) [inline docs/resources/dashboard_share.md:80]: dashboard share

File: docs/resources/dashboard_share.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+- `last_updated` (String) The last time the Dashboard Share was updated
+
+## Import
```
Comment (wigsaparelli) [inline docs/resources/dashboard_share.md:83]: What is this import section?

File: examples/resources/squaredup_dashboard_share/import.sh [hunk 7]
```diff
@@ -0,0 +1,2 @@
+# Shared Dashboard can be imported using the OA ID.
```
Comment (wigsaparelli) [inline examples/resources/squaredup_dashboard_share/import.sh:1]: Dashboard share

File: internal/provider/resource_dashboard_share.go [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+func (r *DashboardShareResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dashboard_share"
+}
+
+func (r *DashboardShareResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Enable Sharing for a dashboard",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the Shared dashboard",
```
Comment (wigsaparelli) [inline internal/provider/resource_dashboard_share.go:51]: dashboard share

**Review by wigsaparelli — APPROVED** [review]
Just some final consistency (terminology and case) comments

File: docs/resources/dashboard_share.md (hunk 6 above)
Comment (shaswot77) [reply docs/resources/dashboard_share.md:83]: [Import](https://developer.hashicorp.com/terraform/cli/import) allows you to import resources that were not created through terraform to its configuration.

**Pull request merged**